src/library/payfine/PayFineControl.java

```diff
@@ -0,0 +1,70 @@
+package library.payfine;
+import library.entities.Library;
+import library.entities.Patron;
+
+public class PayFineControl {
+    
+    private PayFineUI ui;
+    private enum ControlState { INITIALISED, READY, PAYING, COMPLETED, CANCELLED };
+    private ControlState state;
+    
+    private Library library;
+    private Patron patron;
+
+
+    public PayFineControl() {
+        this.library = Library.getInstance();
+        state = ControlState.INITIALISED;
+    }
+    
+    
+    public void setUI(PayFineUI ui) {
+        if (!state.equals(ControlState.INITIALISED)) {
+            throw new RuntimeException("PayFineControl: cannot call setUI except in INITIALISED state");
+        }    
+        this.ui = ui;
+        this.ui.SeTrEaDy();
+        state = ControlState.READY;
+    }
+
+
+    public void cardSwiped(long patronId) {
+        if (!state.equals(ControlState.READY))
+            throw new RuntimeException("PayFineControl: cannot call cardSwiped except in READY state");
+            
+        patron = library.getPatron(patronId);
+        
+        if (patron == null) {
+            ui.DiSplAY("Invalid Patron Id");
+            return;
+        }
+        ui.DiSplAY(patron);
+        ui.SeTpAyInG();
+        state = ControlState.PAYING;
+    }
+    
+    
+    public double payFine(double paymentAmount) {
+        if (!state.equals(ControlState.PAYING)) {
+            throw new RuntimeException("PayFineControl: cannot call payFine except in PAYING state");
+        }
+
+        double change = patron.payFine(paymentAmount);
+        if (change > 0)
+            ui.DiSplAY(String.format("Change: $%.2f", change));
```

Comment:
PR 44 D2 CSG#7 Curly brackets MUST be used

Reply:
[580f0ab](https://github.com/elliotmitchum/ITC205/pull/44/commits/580f0abc014d30709f653bc1435a07f45e2af065)